### PR TITLE
Add usage info to README

### PR DESCRIPTION
**Pull request by lorensr**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,82 @@
-# Temporal proto files  
+# Temporal gRPC API
 
-## How to use
+Proto files describing our gRPC API. Implemented by [Temporal Server](https://github.com/temporalio/temporal) and used by the [SDKs](https://docs.temporal.io/temporal#temporal-sdk) (both Clients and Workers).
```
Comment (Spikhalskiy) [inline README.md:3]: Maybe "Temporal gRPC API"?
And not "describing", but "defining". I wish they described a little more.

File: README.md [hunk 2]
```diff
@@ -1,9 +1,82 @@
...
+  - [Manually](#manually)
+    - [With command line](#with-command-line)
+    - [With a GUI](#with-a-gui)
+- [License](#license)
+
+## Use the API
+
+### With code
+
+Usually you interact with the API via high-level SDK methods like `startWorkflow()`. However, Clients also expose the underlying gRPC services, like:
```
Comment (Spikhalskiy) [inline README.md:19]: Maybe we should use "Service Clients"? "Clients" is a bit of a general term.

File: README.md [hunk 3]
```diff
@@ -1,9 +1,82 @@
...
+#### Generate client stubs
+
+If you're not using an SDK Client (rare), you can generate gRPC client stubs by:
+
+- Adding this repo as a git submodule or subtree inside your repo
+- Generating code in [your language](https://grpc.io/docs/languages/)
+
+### Via REST
+
+See [`temporalio/ui-server`](https://github.com/temporalio/ui-server)
```
Comment (cretz) [inline README.md:34]: I'm not sure we want to promote this as an API server unless it's offering compatibility guarantees. Instead we should be putting HTTP annotations on these protos.

File: README.md [hunk 4]
```diff
@@ -1,9 +1,82 @@
...
+    - [With a GUI](#with-a-gui)
+- [License](#license)
+
+## Use the API
+
+### With code
+
+Usually you interact with the API via high-level SDK methods like `startWorkflow()`. However, Clients also expose the underlying gRPC services, like:
+
+- [`Client.connection.workflowService`](https://typescript.temporal.io/api/classes/client.connection/#workflowservice)
```
Comment (dnr) [inline README.md:21]: Maybe add links to the corresponding way to get a raw grpc client in other languages?

File: README.md (hunk 2 above)
Comment (cretz) [inline README.md:19]: I think it's reasonable to either: 1) put every SDKs approach to raw gRPC invocation in here, or 2) put every SDK's approach to raw gRPC invocation somewhere and link to it from here.

File: README.md [hunk 5]
```diff
@@ -1,9 +1,82 @@
-# Temporal proto files  
+# Temporal gRPC API
 
-## How to use
+Proto files describing our gRPC API. Implemented by [Temporal Server](https://github.com/temporalio/temporal) and used by the [SDKs](https://docs.temporal.io/temporal#temporal-sdk) (both Clients and Workers).
 
-Install as git submodule to the project.
+- [Use the API](#use-the-api)
```
Comment (cretz) [inline README.md:5]: Is this TOC expected to be manually maintained? In GH, the TOC is already available at the top-left of the README.

**Review by cretz — APPROVED** [review]
LGTM, only minor talking points.

**Pull request closed without merging**